### ISSUE TITLE
spdk-18.07 release tarball + dpdk-18.05 release tarball = compiling failure

### ISSUE THREAD

**Issue by beef9999** [spdk/spdk#406]
Download both tarballs from github release site.

## Expected Behavior
Compiling OK

## Current Behavior
Compiling Failure

## Possible Solution
Makefile error, no -fPIC

## Steps to Reproduce
```
# dpdk
make install T=x86_64-native-linuxapp-gcc
make -j

#spdk
./configure --with-dpdk=/dpdk_path/x86_64-native-linuxapp-gcc/
make -j
```

ERROR:

LINK app/iscsi_tgt/iscsi_tgt
/usr/bin/ld: /home/sankuai/rpmbuild/BUILD/spdk-18.07/dpdk/x86_64-native-linuxapp-gcc/lib/librte_eal.a(eal.o): relocation R_X86_64_32 against `.rodata.str1.1' can not be used when making a shared object; recompile with -fPIC
/usr/bin/ld: /home/sankuai/rpmbuild/BUILD/spdk-18.07/dpdk/x86_64-native-linuxapp-gcc/lib/librte_eal.a(eal_hugepage_info.o): relocation R_X86_64_32 against `.rodata' can not be used when making a shared object; recompile with -fPIC

## Context (Environment including OS version, SPDK version, etc.)
CentOS 7.5
spdk-18.07 release tarball + dpdk-18.05 release tarball

**Comment by darsto** [comment]
Since SPDK always builds a shared lib target, you have to compile DPDK as follows:
EXTRA_CFLAGS="-fPIC" make install T=x86_64-native-linuxapp-gcc

We can't really print a nice error message here, but we should definitely mention fPIC in READMEs.

**Comment by wanqunintel** [comment]
@beef9999 you may try with @darsto's method to recompile DPDK and then SPDK. thank you.

**Comment by darsto** [comment]
We no longer build shared libs by default, so the standard compilation should work just fine now.

The patch responsible:
https://review.gerrithub.io/c/spdk/spdk/+/422307/9 (Merged as 0923734 ): build: Conditionally builds single and separate shared libs
